app/src/main/java/com/github/jorgecastillo/kotlinandroid/functional/Future.kt

```diff
@@ -1,35 +1,68 @@
 package com.github.jorgecastillo.kotlinandroid.functional
 
+import kategory.*
 import kotlinx.coroutines.experimental.CommonPool
 import kotlinx.coroutines.experimental.Deferred
 import kotlinx.coroutines.experimental.android.UI
 import kotlinx.coroutines.experimental.async
 import kotlinx.coroutines.experimental.launch
 
+typealias FutureKind<A> = HK<Future.F, A>
+
+fun <A> FutureKind<A>.ev(): Future<A> =
+        this as Future<A>
+
 /**
  * Basic future implementation to achieve asynchronicity based on Kotlin coroutines.
  */
-class Future<T> {
+class Future<T> : FutureKind<T> {
```

Comment:
may be we should append this constructs to an optional extras module inside `kategory`, since Android (non FP familiarized) users will have hard times understanding what's a `***Kind` inside our library for a big while. It would be lovely if we can help them hide it and just import the future, since that simplifies the learning curve to start using this for non familiarized users, A LOT. 

If they want to learn more about FP and how to use the lib, we will probably have docs for that inside of the lib documentation (I think we should), but not force them to understand all the logics and rationales behind FP from day 1.

What do you think?